### ISSUE TITLE
localhost tracing requests

### ISSUE THREAD

**Issue by simao** [openzipkin/zipkin-finagle#71]
We are using zipkin-finagle with the following code:

```
    new SetHostFilter(settings.address.getHostString) andThen Http.client
      .withStreaming(enabled = true)
      .withLabel("director")
      .withTracer(tracer)
      .newService("...")
```

An http request hits our service (device gateway) and in turn contacts other services (director, device-registry, etc), all configured the same way as above. Those services then also support tracing but not through `zipkin-finagle`.

I am trying to make sense of the trace created by this configuration, for example:

```json
[
  {
    "traceId": "b97eb452ffad0ad8",
    "parentId": "b97eb452ffad0ad8",
    "id": "470d90908aeb913f",
    "kind": "SERVER",
    "name": "get",
    "timestamp": 1564048818327015,
    "duration": 2945,
    "localEndpoint": {
      "serviceName": "device-registry"
    },
    "tags": {
      "Finagle-Ctx-com.twitter.finagle.Retries": "0",
      "Host": "ota-device-registry-1-sit",
      "X-B3-ParentSpanId": "b97eb452ffad0ad8",
      "X-B3-Sampled": "true",
      "X-B3-SpanId": "470d90908aeb913f",
      "X-B3-TraceId": "b97eb452ffad0ad8",
      "http.method": "GET",
      "http.path": "/api/v1/devices"
    },
    "shared": true
  },
  {
    "traceId": "b97eb452ffad0ad8",
    "id": "b97eb452ffad0ad8",
    "kind": "SERVER",
    "name": "get",
    "timestamp": 1564048818328000,
    "duration": 13000,
    "localEndpoint": {
      "serviceName": "device-gateway",
      "ipv4": "100.97.84.131",
      "port": 9443
    },
    "remoteEndpoint": {
      "ipv4": "100.115.112.0",
      "port": 54556
    },
    "tags": {
      "srv/finagle.version": "18.6.0"
    }
  },
  {
    "traceId": "b97eb452ffad0ad8",
    "parentId": "b97eb452ffad0ad8",
    "id": "d7ac8f2e5c47ed42",
    "kind": "CLIENT",
    "name": "get",
    "timestamp": 1564048818332000,
    "duration": 9000,
    "localEndpoint": {
      "serviceName": "director",
      "ipv4": "127.0.0.1"
    },
    "remoteEndpoint": {
      "serviceName": "director",
      "ipv4": "100.68.214.1",
      "port": 80
    },
    "annotations": [
      {
        "timestamp": 1564048818332000,
        "value": "ws"
      },
      {
        "timestamp": 1564048818341000,
        "value": "wr"
      }
    ],
    "tags": {
      "clnt/finagle.version": "18.6.0"
    }
  },
  {
    "traceId": "b97eb452ffad0ad8",
    "parentId": "b97eb452ffad0ad8",
    "id": "470d90908aeb913f",
    "kind": "CLIENT",
    "name": "get",
    "timestamp": 1564048818328000,
    "duration": 4000,
    "localEndpoint": {
      "serviceName": "deviceregistry",
      "ipv4": "127.0.0.1"
    },
    "remoteEndpoint": {
      "serviceName": "deviceregistry",
      "ipv4": "100.66.175.42",
      "port": 80
    },
    "annotations": [
      {
        "timestamp": 1564048818328000,
        "value": "ws"
      },
      {
        "timestamp": 1564048818332000,
        "value": "wr"
      }
    ],
    "tags": {
      "clnt/finagle.version": "18.6.0"
    }
  },
  {
    "traceId": "b97eb452ffad0ad8",
    "parentId": "b97eb452ffad0ad8",
    "id": "d7ac8f2e5c47ed42",
    "kind": "SERVER",
    "name": "get",
    "timestamp": 1564048818333014,
    "duration": 8097,
    "localEndpoint": {
      "serviceName": "director"
    },
    "tags": {
      "Accept": "*/*",
      "Finagle-Ctx-com.twitter.finagle.Retries": "0",
      "Host": "ota-director-1-sit",
      "X-B3-ParentSpanId": "b97eb452ffad0ad8",
      "X-B3-Sampled": "true",
      "X-B3-SpanId": "d7ac8f2e5c47ed42",
      "X-B3-TraceId": "b97eb452ffad0ad8",
    },
    "shared": true
  }
]
```

For example, I see a span that has `localEndpoint.serviceName` set to "director" and also `remoteEndpoint.serviceName` set to "director". Is this expected?

This seems to cause zipkin-dependencies to output something strange, as "director" never shows up as a dependency of `device-gateway`.

**Comment by adriancole** [comment]
this looks like #67 and will be "fixed" #68

Basically, there is a modeling problem because we use the service name to identify the local node and finagle uses that to identify the local node or the outbound client. this conflation causes some problems.

If #68 doesn't work for you we can discuss more also.

**Comment by simao** [comment]
Ok, will try once it's out
